### DISCUSSION TITLE
Charge rate in km/h

### DISCUSSION THREAD

**Discussion by mat1990dj** [teslamate-org/teslamate#2280] [Ideas]
When charging at very low rates (12A on 120v) the kW unit is not precise at all since it jumps between 1 and 2.
The tesla API also reports the charge rate in km/h, which can be used to know more precisely the charge rate.
After that if knowing the rated consumption of the car we can extrapolate the actual power feeding the battery and calculate the charge losses.

**Comment by alexeiw123** [comment]
Same metric but for different readons - can we expose charge rate in km/h through MQTT? I'd like to use it in home assistant.

You can calculate it from (charge power(W) / efficiency (Wh/km)), however pack size is not available, so efficiency used would have to be an assumed amount.

**Comment by mat1990dj** [comment]
Not only that, but when actually heating the cabin you can know what amount goes to the battery and what goes to the heating or cooling as that number goes down
Edit: coming from Teslafi I was used to it

**Comment by mat1990dj** [comment]
I’m realizing That the data may be even not be stored in the DB. It could be great to pull it

**Comment by mat1990dj** [comment]
It seems there's even a PR for it https://github.com/adriankumpf/teslamate/pull/2952

**Comment by mat1990dj** [comment]
Hello, would this be something that can be implemented? Is the data pulled or stored anywhere currently?
